source/state_representation/src/geometry/Shape.cpp

```diff
@@ -11,8 +11,16 @@ Shape::Shape(const std::string& name, const std::string& reference_frame) :
   this->set_type(StateType::GEOMETRY_SHAPE);
```

Comment:
On the line above this, you could avoid a casting (slicing) operation. It came to mind because of the new member initialization you did in the default construction.

```diff
--    State(name), center_state_(CartesianPose::Identity(name, reference_frame)) {
++    State(name), center_state_(CartesianState::Identity(name, reference_frame)) {
```